src/aap_eda/api/serializers/auth.py

```diff
@@ -27,37 +29,40 @@ class LoginSerializer(serializers.Serializer):
 # Permissions
 # -----------------------------------------------------
 class PermissionSerializer(serializers.ModelSerializer):
-    resource_type = serializers.CharField(
-        required=True,
-        help_text="Resource type of the permission",
-    )
-
-    action = serializers.CharField(
-        required=True,
-        help_text="Action granted by the permission.",
-    )
-
     class Meta:
         model = models.Permission
         fields = "__all__"
         read_only_fields = ["id"]
 
+    resource_type = serializers.SerializerMethodField()
+    action = serializers.SerializerMethodField()
+
+    def get_resource_type(self, obj):
+        return obj.get_model()
+
+    def get_action(self, obj):
+        return obj.get_action()
+
 
 class PermissionRefSerializer(serializers.ModelSerializer):
     class Meta:
-        model = models.Permission
+        model = models.DABPermission
         fields = ["resource_type", "action"]
-        read_only_fields = ["resource_type", "action"]
+
+    resource_type = serializers.SerializerMethodField()
+    action = serializers.SerializerMethodField()
+
+    def get_resource_type(self, obj):
+        return obj.get_model()
+
+    def get_action(self, obj):
+        return obj.get_action()
 
 
 # -----------------------------------------------------
 # Roles
 # -----------------------------------------------------
 class RoleSerializer(serializers.ModelSerializer):
-    id = serializers.UUIDField(
-        required=True, help_text="Unique UUID of the role"
```

Comment:
I will be putting these id fields back in. I didn't understand what the effect would be at the time.